### ISSUE TITLE
Better warnings for `overfit_batches`

### ISSUE THREAD

**Issue by miccio-dk** [PyTorchLightning/pytorch-lightning#10331]
## Proposed refactoring or deprecation

Currently, enabling `overfit_batches` causes the following warning: 
> UserWarning: You requested to overfit but enabled val/test dataloader shuffling. We are turning it off for you.

 which is logged using `warnings.warn()`. I suggest to use instead `logging.warning()` to improve control of what gets logged on the client side.

### Motivation

According to [this documentation page](https://docs.python.org/3/howto/logging.html):
> warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning
>
> logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted

Thus, by using `warnings.warn()` we implicitly expect the user to disable shuffling themselves, which IMHO defeats the purpose of having such a convenient shortcut ;)

### Pitch

Issue a `logging.warning()` when the user is not expected to act in order to avoid the warning, but rather they are only supposed to take notice of the changes. This may apply to other scenarios as well.  
Furthermore, it would have the added benefit of improving the visualization of logs! 

cc @borda @justusschock @awaelchli @akihironitta @rohitgr7

**Comment by undertherain** [comment]
Most importantly, the logic behind this warning is erroneous. 
In the PL code it is checking if DataLoader has sampler and if sampler is not SequentialSampler. 


I'm using DistributedSampler with `shuffle=False` and of course PL trainer set to not replace distributed sampler, so this warning does not correspond to reality.
